### PR TITLE
get FunctionalTensorWrapper to propagate symints

### PR DESCRIPTION
**Pull request by bdhirsh**
`FunctionalTensorWrapper` needs to know when it's dealing with ints vs. SymInts.

I'm pretty sure the right way to do this is:
```
if (has_symbolic_sizes_strides())
   set_sym_sizes_and_strides(..);
} else {
   set_sizes_and_strides(..);
}
```

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #81302
* __->__ #81301
* #81238

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

### CODE REVIEW

File: aten/src/ATen/FunctionalTensorWrapper.cpp [hunk 1]
```diff
@@ -183,9 +185,20 @@ void FunctionalTensorWrapper::replace_(const Tensor& other) {
...
-    set_sizes_and_strides(value_.sizes(), value_.strides());
+  //
+  if (!has_symbolic_sizes_strides()) {
+    if (sizes() != value_.sizes() || strides() != value_.strides()) {
+      set_sizes_and_strides(value_.sizes(), value_.strides());
+      refresh_numel();
+      refresh_contiguous();
+    }
+  } else {
+    // For sym ints, this will check pointer equality, which I'm pretty sure is what we want.
```
Comment (ezyang) [inline aten/src/ATen/FunctionalTensorWrapper.cpp:196]: yes. Importantly, the pointer equality test will never falsely say that the sizes are the same when they are not

**Review by ezyang — APPROVED** [review]
would be nice to get symbolic numel going. Contiguity is going to be trouble too.

**Review by albanD — COMMENTED** [review]
Can we easily do basic testing for this?

**Comment by zhxchen17** [comment]
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/).

**Pull request closed without merging**